packages/api/src/external/commonwell/patient.ts

```diff
@@ -508,51 +477,211 @@ async function validateCWEnabled({
+      facilityId,
+      getOrgIdExcludeList,
+      requestId,
+      rerunPdOnNewDemographics: false,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.patientDiscovery,
+      properties: {
+        hie: MedicalDataSource.COMMONWELL,
+        patientId: updatedPatient.id,
+        requestId,
+        foundNewDemographicsHere,
+        foundNewDemographicsAcrossHies,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+async function runNexPdIfScheduled({
+  patient,
+  getOrgIdExcludeList,
+  requestId,
+}: {
+  patient: Patient;
+  getOrgIdExcludeList: () => Promise<string[]>;
+  requestId: string;
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const scheduledPdRequest = getCWData(updatedPatient.data.externalData)?.scheduledPdRequest;
+
+  if (scheduledPdRequest) {
+    await update({
+      patient: updatedPatient,
+      facilityId: scheduledPdRequest.facilityId,
+      getOrgIdExcludeList,
+      requestId: scheduledPdRequest.requestId,
+      rerunPdOnNewDemographics: scheduledPdRequest.rerunPdOnNewDemographics,
+    });
+    await resetPatientScheduledPatientDiscoveryRequestId({
+      patient,
+      source: MedicalDataSource.COMMONWELL,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.runScheduledPatientDiscovery,
+      properties: {
+        hie: MedicalDataSource.COMMONWELL,
+        patientId: updatedPatient.id,
+        requestId,
+        scheduledPdRequestId: scheduledPdRequest.requestId,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+async function queryDocsIfScheduled({
+  patient,
+  getOrgIdExcludeList,
+  isFailed = false,
+}: {
+  patient: Pick<Patient, "id" | "cxId">;
+  getOrgIdExcludeList: () => Promise<string[]>;
+  isFailed?: boolean;
+}): Promise<void> {
   const updatedPatient = await getPatientOrFail(patient);
 
   const scheduledDocQueryRequestId = getCWData(
     updatedPatient.data.externalData
   )?.scheduledDocQueryRequestId;
 
   if (scheduledDocQueryRequestId) {
-    const resetPatient = await resetPatientScheduledDocQueryRequestId({
+    if (isFailed) {
+      await setDocQueryProgress({
+        patient: updatedPatient,
+        requestId: scheduledDocQueryRequestId,
+        source: MedicalDataSource.COMMONWELL,
+        downloadProgress: { status: "failed", total: 0 },
+        convertProgress: { status: "failed", total: 0 },
+      });
+    } else {
+      await queryAndProcessDocuments({
+        patient: updatedPatient,
+        requestId: scheduledDocQueryRequestId,
+        getOrgIdExcludeList,
+      });
+    }
+    await resetPatientScheduledDocQueryRequestId({
       patient: updatedPatient,
       source: MedicalDataSource.COMMONWELL,
     });
+  }
+}
 
-    await queryAndProcessDocuments({
-      patient: resetPatient,
-      requestId: scheduledDocQueryRequestId,
-      getOrgIdExcludeList,
+export async function get(
```

Comment:
@leite08 I refactored this to use the name helper functions. One thing to note: you were requesting a patient *without checking* if a personId exists. Was this intentional? It would represent a "partial" state (i.e. there's never supposed to be a patient ID with person ID I think).

The setupUpdate function checks for both, which is used in the refactor. 

Original code:
```
export async function get(
  patient: Patient,
  facilityId: string
): Promise<CommonwellPatient | undefined> {
  const { log } = out(`CW get - M patientId ${patient.id}`);

  const cwEnabled = await validateCWEnabled({
    patient,
    facilityId,
    log,
  });
  if (!cwEnabled) return undefined;

  const cwData = getCWData(patient.data.externalData);
  if (!cwData) return undefined;

  const _initiator = await getCwInitiator(patient, facilityId);
  const initiatorName = _initiator.name;
  const initiatorOid = _initiator.oid;
  const initiatorNpi = _initiator.npi;

  const commonWell = makeCommonWellAPI(initiatorName, addOidPrefix(initiatorOid));
  const queryMeta = organizationQueryMeta(initiatorName, { npi: initiatorNpi });

  return await commonWell.getPatient(queryMeta, cwData.patientId);
}
```

Reply:
Good catch on checking the CW's patient ID.

I don't think we need to check for the person ID, though - there should never be a patient w/ CW patientId and no CW personId, right? We could check just to alert us, but I wouldn't fail the running flow/logic.

nit: any particular reason to move the function from from a section to the file to another? Just asking b/c it removes the history of the function, so it'd be better to leave on the same place, unless there's a good reason to move, of course.

Reply:
wanted to put it next to `remove` as their the two "not epic" CRUD methods, and I found is valuable having create + register right next to each other to read the flow and understand what's happening.

Reply:
re: `don't think we need to check for the person ID, though - there should never be a patient w/ CW patientId and no CW personId, right? We could check just to alert us, but I wouldn't fail the running flow/logic.`

The `setupPatient` does all three: (1) missing external data, (2) missing patientId or (3) missing personId. Is this not desired?

Reply:
Not an issue, I was mostly:
1. aligning on needs/requirements;
1. discussing if we should fail if a person ID is missing when we want to DQ.

Reply:
@leite08 please resolve if you're cool with the refactor.